### ISSUE TITLE
link tracking function is trimming long urls to 255 char

### ISSUE THREAD

**Issue by hadifarnoud** [atech/postal#668]
we have long urls which contains token key for download. when I click on the url it redirects to the original one which is trimmed to 255 characters.

**Comment by willpower232** [comment]
You'll have to modify the database for that specific Postal server to alter the `varchar` field for `url` into a `text` field (or `mediumtext` etc)

![image](https://user-images.githubusercontent.com/1619102/46092361-a0233e00-c1ac-11e8-9f98-26faec63cbbe.png)

Atech or someone else will have to add a migration to make a similar change. I'm not sure how to migrate those databases.

**Comment by hadifarnoud** [comment]
I think we can consider this a bug. there are many cases that urls are longer than 255 characters.

**Comment by willpower232** [comment]
Yes it most certainly is but the only way to fix it is to make the database modifications I have mentioned.

**Comment by hadifarnoud** [comment]
I was unable to fix it locally. Not a Rails literate here. I ran into gem install issues and eventually gave up after days of debugging. 

can someone please make a pull request for this bug?

**Comment by willpower232** [comment]
As I said, the problem is in the database so you'll need to use MySQL tools to make the fix for now.

**Comment by willpower232** [comment]
I've found a fix but it needs to be merged and run manually at the minute
